src/hooks/useDocumentDrives.ts

```diff
@@ -6,41 +6,35 @@ import { useMemo } from 'react';
 
 // map of DocumentDriveServer objects and their Document Drives
 const documentDrivesAtom = atom(
-    new Map<IDocumentDriveServer, DocumentDriveDocument[]>()
+    new Map<IDocumentDriveServer, DocumentDriveDocument[]>(),
 );
 
 // creates a derived atom that encapsulates the Map of Document Drives
-const readWriteDocumentDrivesAtom = (server: IDocumentDriveServer) =>
-    useMemo(
-        () =>
-            atom(
-                get => get(documentDrivesAtom).get(server) ?? [],
-                (_get, set, newDrives: DocumentDriveDocument[]) => {
-                    set(documentDrivesAtom, map =>
-                        new Map(map).set(server, newDrives)
-                    );
-                }
-            ),
-        [server]
+const readWriteDocumentDrivesAtom = (server: IDocumentDriveServer) => () =>
```

Comment:
moved useMemo call into hook  because hooks should only be called inside other hooks